runtime/binding-grpc-kafka/src/main/java/io/aklivity/zilla/runtime/binding/grpc/kafka/internal/stream/GrpcKafkaProxyFactory.java

```diff
@@ -1388,6 +1408,39 @@ private void doGrpcBegin(
                 traceId, authorization, affinity, extension);
         }
 
+        private void doGrpcBegin(
+            long traceId,
+            long authorization,
+            long affinity,
+            Array32FW<KafkaHeaderFW> headers)
+        {
+            state = GrpcKafkaState.openingReply(state);
+
+            Array32FW.Builder<GrpcMetadataFW.Builder, GrpcMetadataFW> builder =
+                grpcMetadataRW.wrap(metaBuffer, 0, metaBuffer.capacity());
+
+            headers.forEach(h ->
+            {
+                if (META_PREFIX.equals(h.name().value().getStringWithoutLengthUtf8(0, 5)))
```

Comment:
What if header name length is less than 5 bytes?

Lets compare the `meta:` prefix via direct buffer instead of `String` to avoid the need to create the `String` just for the comparison.